Varta: add battery control

zoomy01 opened the pull request:
https://github.com/evcc-io/evcc/discussions/19631

zoomy01 committed:
Update varta.yaml

https://github.com/evcc-io/evcc/discussions/19631

zoomy01 committed:
Update varta.yaml

VolkerK62 commented:
Vorschlag: 
den Parameter `dischargepower`  als positiven Wert ausführen und das entsprechende Scaling innerhalb batterymode machen.
Ich denke, das ist intuitiver.

zoomy01 committed:
Update varta.yaml

Parameter dischargepower in positiven Wert geändert, Umwandlung erfolgt im Template

zoomy01 commented:
den Parameter dischargepower bräuchte es innerhalb von evvc eigentlich nicht, da er für nicht genutzt wird.

Gibt es eine Möglichkeit den Ist-Wert aus Register 1074 zu irgendwo zu speichern, wenn hold aktiviert wird und beim Wechsel auf normal diesen Wert wieder zurück zu schreiben?

andig commented:
Nein, gibts nicht :/

andig commented on templates/definition/meter/varta.yaml:
```diff
@@ -58,5 +60,36 @@ render: |
+          source: sequence
+          set:
+          - source: const
+            value: {{ if lt .dischargepower 0 }}{{ .dischargepower }}{{ else }}{{ mul -1 .dischargepower }}{{ end }} # W
```
Die Logik brauchts egtl. nicht, negative Lesitungen gibts nirgendwo in evcc.

zoomy01 replied:
ok

andig commented on templates/definition/meter/varta.yaml:
```diff
@@ -23,6 +23,8 @@ params:
     default: 502
   - name: capacity
     advanced: true
```
Konsequenterweise müsste das `maxdischargepower` heissen, gerne mit Beschreibung in die defaults.yaml einfügen.

andig commented on templates/definition/meter/varta.yaml:
```diff
@@ -58,5 +60,36 @@ render: |
+                address: 1074  # discharge power
+                type: writeholding
+                decode: int16
+      - case: 2  # hold
```
Es fehlt dann noch case 3, der kann `ErrNotAvailable` zurück geben.

zoomy01 committed:
Update varta.yaml

Case 3 ergänzt, Parameter umbenannt

zoomy01 committed:
Update defaults.yaml

maxdischargepower ergänzt

zoomy01 committed:
Update varta.yaml

watchdog ergänzt

zoomy01 committed:
Update varta.yaml

sourcery-ai comments

andig commented:
Was fehlt hier noch?

zoomy01 reviewed (commented):
great

zoomy01 committed:
Update varta.yaml

small correction

zoomy01 committed:
Update varta.yaml

unit: W for maxdischargepower

zoomy01 commented:
> Was fehlt hier noch?

von meiner Seite nicht. ;)

andig committed:
Merge branch 'master' into patch-2
